server/src/test/java/org/elasticsearch/search/SearchModuleTests.java

```diff
@@ -607,4 +612,73 @@ public String getWriteableName() {
             return "test";
         }
     }
+
+    static class CompatQueryBuilder extends AbstractQueryBuilder<CompatQueryBuilder> {
+        public static final String NAME = "compat_name";
+        public static final ParseField NAME_V7 = new ParseField(NAME)
```

Comment:
Very minor nit here: the way this test is written will work as master goes to 8.0.0 and beyond, so the `_V7` here, though currently accurate, will likely eventually become misleading. Perhaps `NAME_OLD` or something else that doesn't have an explicit version number?

Reply:
good point, will fix it